File: app/src/main/java/com/hjc/base/widget/MyScrollView.java
package com.hjc.base.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.ScrollView;

import com.blankj.utilcode.util.LogUtils;

public class MyScrollView extends ScrollView {
    public MyScrollView(Context context) {
        super(context);
    }

    public MyScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MyScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        boolean result = super.onInterceptTouchEvent(ev);
        LogUtils.e("MyScrollView: onInterceptTouchEvent: " + result);
        return result;
    }
}

File: module-senior/src/main/java/com/hjc/module_senior/widget/event/MyScrollView.kt
package com.hjc.module_senior.widget.event

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.ScrollView
import com.blankj.utilcode.util.LogUtils

class MyScrollView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : ScrollView(context, attrs, defStyleAttr) {

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        val result = super.onInterceptTouchEvent(ev)
        LogUtils.e("MyScrollView: onInterceptTouchEvent: $result")
        return result
    }
}